### PR TITLE
Switch set_search_options from taking a dict to kwargs

### PR DESCRIPTION
**Pull request by RyanUnderhill**
Now 20% more pythonic!

### CODE REVIEW

**Review by baijumeswani — APPROVED** [review]

File: src/python/python.cpp [hunk 1]
```diff
@@ -94,7 +94,7 @@ struct PyGeneratorParams {
     }
   }
 
-  void SetSearchOptions(const pybind11::dict& dict) {
+  void SetSearchOptions(const pybind11::kwargs& dict) {
```
Comment (yufenglee) [inline src/python/python.cpp:97]: could we list out all the valid search options here:
https://github.com/microsoft/onnxruntime-genai/blob/18adb67dce145579779dd0b6a3e2b25826a7b459/src/python/python.cpp#L182

Comment (RyanUnderhill) [reply src/python/python.cpp:97]: Do it as python help on that function? We should provide a url since the options match the config documentation.

Comment (yufenglee) [reply src/python/python.cpp:97]: i mean add a comment here

**Review by baijumeswani — APPROVED** [review]

File: src/python/python.cpp (hunk 1 above)
Comment (RyanUnderhill) [reply src/python/python.cpp:97]: Done

**Pull request merged**